modules/imgcodecs/src/grfmt_jpeg.cpp

```diff
@@ -687,6 +688,27 @@ bool JpegEncoder::write( const Mat& img, const std::vector<int>& params )
                 rst_interval = params[i+1];
                 rst_interval = MIN(MAX(rst_interval, 0), 65535L);
             }
+
+            if( params[i] == IMWRITE_JPEG_SAMPLING_FACTOR )
+            {
+                sampling_factor = static_cast<uint32_t>(params[i+1]);
+
+                switch ( sampling_factor )
+                {
+                    case IMWRITE_JPEG_SAMPLING_FACTOR_411:
+                    case IMWRITE_JPEG_SAMPLING_FACTOR_420:
+                    case IMWRITE_JPEG_SAMPLING_FACTOR_422:
+                    case IMWRITE_JPEG_SAMPLING_FACTOR_440:
+                    case IMWRITE_JPEG_SAMPLING_FACTOR_444:
+                    // OK.
+                    break;
+
+                    default:
+                    // Unknown parameter is set.
+                    sampling_factor = 0;
```

Comment:
It makes sense to emit warning here through `CV_LOG_WARNING(NULL, "Unknown value for IMWRITE_JPEG_SAMPLING_FACTOR: " << sampling_factor);`

Reply:
Thank you for your comment, i fixed it.